components/use-messages/message-item.tsx

```diff
@@ -0,0 +1,148 @@
+  ({ className, id, text, delay, destroy, onClose, closeable, ...rest }) => {
+    const theme = useTheme()
+    const { color, bgColor, icon } = useMemo(() => getStyles(theme.palette, rest.color), [
+      theme.palette,
+      rest.color,
+    ])
+    let boxShadow: string = theme.expressiveness.D2
+    if (rest.color !== 'default' && !rest.shadow) {
+      boxShadow = 'none'
+    }
+    const [visible, setVisible] = useState<boolean>(false)
+    const [hide, setHide] = useState<boolean>(false)
+    const [hover, setHover] = useState<boolean>(false)
+    useEffect(() => {
+      const timer = setTimeout(() => {
+        setVisible(true)
+        clearTimeout(timer)
+      }, 10)
+      return () => clearTimeout(timer)
+    }, [])
+    useEffect(() => {
+      let timer: any = null
+      if (hover) {
+        clearTimeout(timer)
+      } else {
+        if (delay) {
+          timer = setTimeout(() => {
+            setHide(true)
+            clearTimeout(timer)
+          }, delay)
+        }
+      }
+      return () => {
+        delay && clearTimeout(timer)
+      }
+    }, [delay, hover])
+    useEffect(() => {
+      let timer: any = null
+      if (hide) {
+        timer = setTimeout(() => {
+          destroy && destroy(id)
+          onClose && onClose(id)
+          clearTimeout(timer)
+        }, transitionDuration)
+      }
+      return () => clearTimeout(timer)
+    }, [hide])
+    const handleClose = () => {
+      setHide(true)
+    }
+    const handleMouseEnter = () => {
+      setHover(true)
+    }
+    const handleMouseLeave = () => {
+      setHover(false)
+    }
+    return (
+      <div
+        key={id}
+        className={`message ${className} ${visible ? 'visible' : ''} ${hide ? 'hide' : ''}`}
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handleMouseLeave}>
+        <div className="icon">{rest.icon || icon}</div>
+        <div className="text">{text}</div>
+        {closeable && (
+          <div className="close" onClick={handleClose}>
+            <X />
+          </div>
+        )}
+        <style jsx>{`
+          .message {
+            max-width: 71.4286rem;
+            background-color: ${bgColor};
+            color: ${color};
+            border: 0;
+            border-radius: ${theme.expressiveness.R2};
+            padding: ${theme.layout.gap};
+            box-shadow: ${boxShadow};
+            margin-top: calc(${theme.layout.gap} * 1.5);
+            transform: translate(0, -100%);
+            opacity: 0;
+            transition: transform ${transitionDuration}ms, opacity ${transitionDuration}ms,
+              margin-top ${transitionDuration}ms;
+            display: flex;
+          }
+          .message:first-child {
+            margin-top: 0;
+          }
+          .message.visible {
+            transform: translate(0, 0);
+            opacity: 1;
+            z-index: 2;
+          }
+          .message.hide {
+            margin-top: -3.5714rem;
```

Comment:
Why not use the layout.gap variable here

Reply:
It is message component height value, not gap value, should not attract with gap config.